src/main/java/com/conveyal/datatools/editor/controllers/api/FeedInfoController.java

```diff
@@ -92,162 +77,83 @@ public static Object createFeedInfo(Request req, Response res) {
-
+        GlobalTx gtx = null;
         try {
             feed = Base.mapper.readValue(req.body(), EditorFeed.class);
-            GlobalTx gtx = VersionedDataStore.getGlobalTx();
+            gtx = VersionedDataStore.getGlobalTx();
 
             if(!gtx.feeds.containsKey(feed.id)) {
-                gtx.rollback();
                 halt(400);
-                return null;
             }
 
             gtx.feeds.put(feed.id, feed);
             gtx.commit();
 
-            return Base.toJson(feed, false);
+            return feed;
         } catch (HaltException e) {
             LOG.error("Halt encountered", e);
             throw e;
         } catch (Exception e) {
             e.printStackTrace();
             halt(400);
+        } finally {
+            if (gtx != null) gtx.rollbackIfOpen();
         }
         return null;
     }
 
-    public static void applyJsonToFeedInfo(EditorFeed source, String json) throws IOException {
-        ObjectMapper mapper = new ObjectMapper();
-        JsonNode node = mapper.readTree(json);
-        Iterator<Map.Entry<String, JsonNode>> fieldsIter = node.fields();
-        while (fieldsIter.hasNext()) {
-            Map.Entry<String, JsonNode> entry = fieldsIter.next();
-
-            if (entry.getValue().isNull()) {
-                continue;
-            }
-
-            if(entry.getKey().equals("color")) {
-                source.color = entry.getValue().asText();
-            }
-
-            if(entry.getKey().equals("defaultLat")) {
-                source.defaultLat = entry.getValue().asDouble();
-            }
-
-            if(entry.getKey().equals("defaultLon")) {
-                source.defaultLon = entry.getValue().asDouble();
-            }
-
-            if(entry.getKey().equals("routeTypeId")) {
-                source.defaultRouteType = GtfsRouteType.fromGtfs(entry.getValue().asInt());
-            }
-
-            if(entry.getKey().equals("feedPublisherName")) {
-                source.feedPublisherName = entry.getValue().asText();
-            }
-
-            if(entry.getKey().equals("feedVersion")) {
-                source.feedVersion = entry.getValue().asText();
-            }
-
-            if(entry.getKey().equals("feedPublisherUrl")) {
-                String url = entry.getValue().asText();
-                try {
-                    source.feedPublisherUrl = new URL(url);
-
-                } catch (MalformedURLException e) {
-                    halt(400, "URL '" + url + "' not valid.");
-                }
-                source.feedPublisherUrl = new URL(entry.getValue().asText());
-            }
-
-            if(entry.getKey().equals("feedLang")) {
-                source.feedLang = entry.getValue().asText();
-            }
+    /**
+     * Delete NOT ONLY feed info, but also entirely delete the record of the feed in the editor database.
+     * @param req
+     * @param res
+     * @return
+     */
+    public static Object deleteFeedInfo(Request req, Response res) {
```

Comment:
Should this function be called something that makes it more clear what it does in case future @mattwigway doesn't read the javadocs and assumes it deletes feed info? Something like `wipeAllTraceOfFeedFromDatabaseIHopeYouMakeBackups`?